### PR TITLE
Add docs for Microsoft.AspNetCore.Http

### PR DESCRIPTION
**Pull request by pranavkm**

### CODE REVIEW

File: src/Http/Http.Features/src/IHttpRequestFeature.cs [hunk 1]
```diff
@@ -12,46 +12,56 @@ namespace Microsoft.AspNetCore.Http.Features
...
         /// is in its original form, without un-escaping.
         /// </summary>
         string QueryString { get; set; }
 
         /// <summary>
-        /// The request target as it was sent in the HTTP request. This property contains the
+        /// Gets or sets the request target as it was sent in the HTTP request.
+        /// <para>
+        /// This property contains the
         /// raw path and full query, as well as other request targets such as * for OPTIONS
```
Comment (Tratcher) [inline src/Http/Http.Features/src/IHttpRequestFeature.cs:62]: Can you fix the un-even lines here?

File: src/Http/Http.Features/src/IHttpRequestFeature.cs [hunk 2]
```diff
@@ -60,17 +70,24 @@ public interface IHttpRequestFeature
         string RawTarget { get; set; }
 
         /// <summary>
-        /// Headers included in the request, aggregated by header name. The values are not split
+        /// Gets or sets headers included in the request, aggregated by header name.
+        /// <para>
+        /// The values are not split
         /// or merged across header lines. E.g. The following headers:
```
Comment (Tratcher) [inline src/Http/Http.Features/src/IHttpRequestFeature.cs:76]: Uneven

File: src/Http/Http.Features/src/IHttpResponseFeature.cs [hunk 3]
```diff
@@ -13,37 +13,43 @@ namespace Microsoft.AspNetCore.Http.Features
     public interface IHttpResponseFeature
     {
         /// <summary>
-        /// The status-code as defined in RFC 7230. The default value is 200.
+        /// Gets or sets the status-code as defined in RFC 7230..
```
Comment (Tratcher) [inline src/Http/Http.Features/src/IHttpResponseFeature.cs:16]:
```suggestion
        /// Gets or sets the status-code as defined in RFC 7230.
```

File: src/Http/Http/src/BindingAddress.cs [hunk 4]
```diff
@@ -7,6 +7,9 @@
 
 namespace Microsoft.AspNetCore.Http
 {
+    /// <summary>
+    /// An address that a HTTP server is bound to.
```
Comment (Tratcher) [inline src/Http/Http/src/BindingAddress.cs:11]:
```suggestion
    /// An address that a HTTP server may bind to.
```

**Review by Tratcher — APPROVED** [review]

File: src/Http/Http.Features/src/IFormFeature.cs [hunk 5]
```diff
@@ -14,12 +17,22 @@ public interface IFormFeature
         bool HasFormContentType { get; }
 
         /// <summary>
-        /// The parsed form, if any.
+        /// The parsed form.
```
Comment (JamesNK) [inline src/Http/Http.Features/src/IFormFeature.cs:20]:
```suggestion
        /// Gets or sets the parsed form.
```

**Review by JamesNK — APPROVED** [review]

File: src/Http/Http.Features/src/IHttpRequestFeature.cs [hunk 6]
```diff
@@ -12,46 +12,56 @@ namespace Microsoft.AspNetCore.Http.Features
...
         /// </summary>
         string QueryString { get; set; }
 
         /// <summary>
-        /// The request target as it was sent in the HTTP request. This property contains the
+        /// Gets or sets the request target as it was sent in the HTTP request.
+        /// <para>
+        /// This property contains the
         /// raw path and full query, as well as other request targets such as * for OPTIONS
         /// requests (https://tools.ietf.org/html/rfc7230#section-5.3).
```
Comment (pranavkm) [inline src/Http/Http.Features/src/IHttpRequestFeature.cs:63]:
```suggestion
        /// This property contains the raw path and full query, as well as other request targets
        /// such as * for OPTIONS requests (https://tools.ietf.org/html/rfc7230#section-5.3).
```

File: src/Http/Http.Features/src/IHttpRequestFeature.cs (hunk 2 above)
Comment (pranavkm) [inline src/Http/Http.Features/src/IHttpRequestFeature.cs:76]:
```suggestion
        /// The values are not split or merged across header lines. E.g. The following headers:
```

**Pull request merged**